Tools/CMake/AMReXOptions.cmake

```diff
@@ -0,0 +1,323 @@
+###############################################
+
+# Here we define the default config options   #
+# that can be overwritten by the user         #
+
+###############################################
+include_guard(GLOBAL)
+
+include(CMakeDependentOption)
+
+#
+# Define a macro to print active options
+#
+macro (print_option _var)
+   if (${_var})
+      message( STATUS "   ${_var}")
+   endif ()
+endmacro ()
+
+
+#
+# Dimensionality of the build  ===============================================
+#
+set(AMReX_SPACEDIM_VALUES 1 2 3)
+set(AMReX_SPACEDIM 3 CACHE STRING "Dimension of AMReX build: <1,2,3>")
+set_property(CACHE AMReX_SPACEDIM PROPERTY STRINGS ${AMReX_SPACEDIM_VALUES})
+if(NOT AMReX_SPACEDIM IN_LIST AMReX_SPACEDIM_VALUES)
+   message(FATAL_ERROR "AMReX_SPACEDIM=${AMReX_SPACEDIM} is not allowed."
+      " Must be one of ${AMReX_SPACEDIM_VALUES}")
+endif()
+message( STATUS "Building AMReX with AMReX_SPACEDIM = ${AMReX_SPACEDIM}")
+
+#
+# Populate the cache and check the value of the user-definable options
+#
+message(STATUS "Configuring AMReX with the following options enabled: ")
+
+#
+# This is the option to enable/disable xSDK mode
+#
+# To handle both XDSK options and correponding plain AMReX options,
+# we make use of policy CMP0077 introduced as a default in CMake 3.13
+# Under policy CMP0077, normal variables prevent option()
+# to set internal variables of the same name.
+# Example: if XSDK mode is ON and XSDK_ENABLE_Fortran=ON, a normal
+# variable AMReX_FORTRAN will be created and set to ON.
+# This will stop the subsequent option( AMReX_FORTRAN "Enable Fortran language" ON )
+# from being executed and no entry AMReX_FORTRAN will be created in the cache
+#
+option( USE_XSDK_DEFAULTS "Enable xSDK mode"  OFF )
+print_option( USE_XSDK_DEFAULTS )
+
+
+#
+# Option to control the type of library: static vs shared  ===================
+#
+if ( USE_XSDK_DEFAULTS )
+   option( BUILD_SHARED_LIBS "Build AMReX shared library" ON )
+else ()
+   option( BUILD_SHARED_LIBS "Build AMReX shared library" OFF )
+endif ()
+print_option( BUILD_SHARED_LIBS )
+
+
+#
+# Option to control if Fortran must be enabled  ==============================
+#
+if ( USE_XSDK_DEFAULTS )
+   option( XSDK_ENABLE_Fortran "Enable Fortran language" OFF )
+   set( AMReX_FORTRAN ${XSDK_ENABLE_Fortran} )
+   print_option(XSDK_ENABLE_Fortran)
+endif ()
+
+option( AMReX_FORTRAN "Enable Fortran language" ON )
+print_option( AMReX_FORTRAN )
+
+#
+# Option to control precision of the build  ==================================
+#
+if ( USE_XSDK_DEFAULTS )
+   set( XSDK_PRECISION "DOUBLE" CACHE STRING "Precision:<SINGLE,DOUBLE>" )
+   set( AMReX_PRECISION ${XSDK_PRECISION})
+   set( AMReX_PARTICLES_PRECISION ${XSDK_PRECISION})
+   print_option( XSDK_PRECISION )
+endif ()
+
+
+set(AMReX_PRECISION_VALUES SINGLE DOUBLE)
```

Comment:
Just a minor taste quesion:
Since this is not an abbreviation of brand name, I picked lowercase `single`/`double` last when I added this to WarpX. I am ok to go all caps, too, just slightly less pretty.

Reply:
I picked upper case because the options names are upper case anyways. I can make it case-insensitive tho.

Reply:
Oh, case insensitive is nice for this one, too. Only if it's not too much work.

Reply:
Actually, making it case-insensitive requires a little bit of extra logic. Also, XSDK seems to prescribe the use of upper case.At this point, I will leave it upper case. Sorry :-(